app/models/concerns/attachmentable.rb

```diff
@@ -43,7 +43,7 @@ def check_image_dimensions
 
       width, height = FastImage.size(attachment.queued_for_write[:original].path)
 
-      raise Mastodon::DimensionsValidationError, "#{width}x#{height} images are not supported" if width.present? && height.present? && (width * height >= MAX_MATRIX_LIMIT)
+      raise Mastodon::DimensionsValidationError, "#{width}x#{height} images are not supported, w*h must be under #{MAX_MATRIX_LIMIT}" if width.present? && height.present? && (width * height >= MAX_MATRIX_LIMIT)
```

Comment:
There has to be a more readable way to communicate the matrix limit than `w*h`. Perhaps: `...images are not supported, must be below #{MAX_MATRIX_LIMIT} sqpx` (square pixels)?

Reply:
either is an improvement, updated as suggested.